src/ImageData/StokesFilesConnector.cc

```diff
@@ -0,0 +1,324 @@
+                err = "Different file types can not be concatenated!\n";
+                return false;
+            }
+
+            // get the common file name start from the head
+            int tmp_pos_head = StrCmp(prefix_file_name, stokes_file.file());
+            if (tmp_pos_head < pos_head) {
+                pos_head = tmp_pos_head;
+            }
+            prefix_file_name = prefix_file_name.substr(0, pos_head);
+
+            // get the common file name start from the tail
+            std::string tmp_reverse_filename = stokes_file.file();
+            std::reverse(tmp_reverse_filename.begin(), tmp_reverse_filename.end());
+            int tmp_pos_tail = StrCmp(postfix_file_name, tmp_reverse_filename);
+            if (tmp_pos_tail < pos_tail) {
+                pos_tail = tmp_pos_tail;
+            }
+            postfix_file_name = postfix_file_name.substr(0, pos_tail);
+        }
+    }
+
+    // set the final name of concatenated image
+    std::reverse(postfix_file_name.begin(), postfix_file_name.end());
+    _concatenate_name = prefix_file_name + _concatenate_name + postfix_file_name;
+
+    return true;
+}
+
+bool StokesFilesConnector::StokesFilesValid(std::string& err, int& stokes_axis) {
+    if (_loaders.size() < 2) {
+        err = "Need at least two files to concatenate!\n";
+        return false;
+    }
+
+    casacore::IPosition ref_shape(0);
+    int ref_spectral_axis = -1;
+    int ref_stokes_axis = -1;
+    int ref_index = 0;
+
+    for (auto& loader : _loaders) {
+        casacore::IPosition shape;
+        int spectral_axis;
+
+        if (ref_index == 0) {
+            loader.second->FindCoordinateAxes(ref_shape, ref_spectral_axis, ref_stokes_axis, err);
+        } else {
+            loader.second->FindCoordinateAxes(shape, spectral_axis, stokes_axis, err);
+            if ((ref_shape.nelements() != shape.nelements()) || (ref_shape != shape) || (ref_spectral_axis != spectral_axis) ||
+                (ref_stokes_axis != stokes_axis)) {
+                err = "Images shapes or axes are not consistent!\n";
+                return false;
+            }
+        }
+        ++ref_index;
+    }
+    return true;
+}
+
+bool StokesFilesConnector::GetStokesType(const CARTA::StokesType& in_stokes_type, casacore::Stokes::StokesTypes& out_stokes_type) {
+    bool success(false);
+    switch (in_stokes_type) {
+        case CARTA::StokesType::I:
+            out_stokes_type = casacore::Stokes::I;
+            success = true;
+            break;
+        case CARTA::StokesType::Q:
+            out_stokes_type = casacore::Stokes::Q;
+            success = true;
+            break;
+        case CARTA::StokesType::U:
+            out_stokes_type = casacore::Stokes::U;
+            success = true;
+            break;
+        case CARTA::StokesType::V:
+            out_stokes_type = casacore::Stokes::V;
+            success = true;
+            break;
+        default:
+            break;
+    }
+    return success;
+}
```

Comment:
This can also be simplified with an inline `unordered_map` to map the CARTA types to the casacore types.

Reply:
I think just simplifying to remove `success` and use `return true` or `return false` would make it nice and compact